pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -303,7 +304,11 @@ def _register_ddp_hooks(self) -> None:
     def configure_ddp(self):
         self.pre_configure_ddp()
         self._model = DistributedDataParallel(
-            LightningDistributedModule(self.model), device_ids=self.determine_ddp_device_ids(), **self._ddp_kwargs
+            LightningDistributedModule(self.model)
+            if self.lightning_module.trainer.state.fn == TrainerFn.FITTING
+            else self.model,
+            device_ids=self.determine_ddp_device_ids(),
+            **self._ddp_kwargs,
         )
```

Comment:
prefer early returning with something like this:
```
# skip wrapping the model if we're not fitting as no gradients need to be exchanged
trainer_fn = self.lightning_module.trainer.state.fn
if trainer_fn != TrainerFn.FITTING:
    rank_zero_debug(f"In {trainer_fn}: Skipping wrapping the model with DistributedDataParallel") 
    return
self.pre_configure_ddp()
self._model = DistributedDataParallel(....)
```

Reply:
@SeanNaren - do you think we should (re)set self._model here to self.lightning_module if used for evaluation or prediction?

Reply:
@ananthsub For evaluation/prediction, do we want to make sure self._model is LightningDistributedModule?